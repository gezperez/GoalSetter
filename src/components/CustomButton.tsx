import React, { FC, memo } from 'react'
import { StyleProp, StyleSheet, Text, TextStyle, TouchableOpacity, ViewStyle } from 'react-native'

import { Colors } from '~/utils'
import Typography from '~/utils/Typographys'

type CustomButtonProps = {
  label: string
  onPress?: () => void
  style?: StyleProp<ViewStyle>
  accessibilityLabel?: string
  testID?: string
}

const CustomButton: FC<CustomButtonProps> = ({ label, onPress, style, accessibilityLabel, testID }) => (
  <TouchableOpacity
    style={[styles.container, style]}
    onPress={onPress}
    disabled={!onPress}
    accessibilityLabel={accessibilityLabel}
    testID={testID}
  >
    <Text style={styles.label}>{label}</Text>
  </TouchableOpacity>
)

type Styles = {
  container: ViewStyle
  label: TextStyle
}

const styles = StyleSheet.create<Styles>({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
    marginHorizontal: 50,
    backgroundColor: Colors.BUTTON,
    borderRadius: 5,
  },
  label: {
    ...Typography.button,
  },
})

export default memo(CustomButton)
