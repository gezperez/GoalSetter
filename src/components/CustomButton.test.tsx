import React from 'react'
import { StyleSheet, ViewStyle } from 'react-native'
import { fireEvent, render } from '@testing-library/react-native'

import CustomButton from './CustomButton'

describe('CustomButton', () => {
  afterEach(jest.resetAllMocks)

  it('should render with required props', () => {
    const { toJSON } = render(<CustomButton label="button label" />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render with non required props', () => {
    const { toJSON } = render(<CustomButton label="button label" style={styles.button} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render and support touchable props', () => {
    const mockPressHandler = jest.fn()

    const { toJSON, getByLabelText } = render(
      <CustomButton label="button label" onPress={mockPressHandler} accessibilityLabel="button" testID="button" />
    )

    fireEvent.press(getByLabelText('button'))

    expect(mockPressHandler).toBeCalled()
    expect(toJSON()).toMatchSnapshot()
  })
})

type Styles = {
  button: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  button: {
    margin: 15,
  },
})
