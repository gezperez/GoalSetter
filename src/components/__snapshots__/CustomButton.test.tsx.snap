// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton should render and support touchable props 1`] = `
<View
  accessibilityLabel="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5FBA20",
      "borderRadius": 5,
      "justifyContent": "center",
      "marginHorizontal": 50,
      "opacity": 1,
      "padding": 20,
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 12,
        "lineHeight": 14,
      }
    }
  >
    button label
  </Text>
</View>
`;

exports[`CustomButton should render with non required props 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5FBA20",
      "borderRadius": 5,
      "justifyContent": "center",
      "margin": 15,
      "marginHorizontal": 50,
      "opacity": 1,
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 12,
        "lineHeight": 14,
      }
    }
  >
    button label
  </Text>
</View>
`;

exports[`CustomButton should render with required props 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5FBA20",
      "borderRadius": 5,
      "justifyContent": "center",
      "marginHorizontal": 50,
      "opacity": 1,
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 12,
        "lineHeight": 14,
      }
    }
  >
    button label
  </Text>
</View>
`;
